### PR TITLE
tweak the max depth deprecation

### PR DESCRIPTION
**Pull request by xabbuh**

### CODE REVIEW

**Comment by Ener-Getick** [comment]
:+1: BTW i forgot to create an upgrade file for 2.1 and 3.0.

**Comment by Ener-Getick** [comment]
Status: reviewed

File: Context/Context.php [hunk 1]
```diff
@@ -181,12 +181,12 @@ public function setGroups(array $groups)
...
      *
-     * @deprecated since 2.1, to be removed in 3.0. Use {@link self::enableMaxDepth()} instead.
+     * @deprecated since 2.1, to be removed in 3.0. Use {@link self::enableMaxDepth()} and {@link self::disableMaxDepth()} instead.
      */
-    public function setMaxDepth($maxDepth, $triggerDeprecation = true)
+    public function setMaxDepth($maxDepth)
     {
-        if ($triggerDeprecation) {
-            @trigger_error(sprintf('%s is deprecated since version 2.1 and will be removed in 3.0. Use %s::enabledMaxDepth() instead.', __METHOD__, __CLASS__), E_USER_DEPRECATED);
+        if (0 === func_num_args() || func_get_arg(1)) {
```
Comment (Ener-Getick) [inline Context/Context.php]: should be ```1 >= func_num_args() || func_get_arg(1)```

Comment (xabbuh) [inline Context/Context.php]: Actually, it should be `1 >= func_num_args() || func_get_arg(1)`, shouldn't it?

File: Context/Context.php [hunk 2]
```diff
@@ -200,21 +200,25 @@ public function setMaxDepth($maxDepth, $triggerDeprecation = true)
      *
      * @deprecated since version 2.1, to be removed in 3.0. Use {@link self::isMaxDepthEnabled()} instead.
      */
-    public function getMaxDepth($triggerDeprecation = true)
+    public function getMaxDepth()
     {
-        if ($triggerDeprecation) {
+        if (0 === func_num_args() || func_get_arg(1)) {
```
Comment (xabbuh) [inline Context/Context.php]: fixed the index here

**Comment by xabbuh** [comment]
Hm, need to check why tests are failing.

**Comment by xabbuh** [comment]
Okay, tests finally pass.

**Comment by Ener-Getick** [comment]
Thank you @xabbuh !

**Pull request merged**